packages/web-react/src/types/modalComposed.ts

```diff
@@ -0,0 +1,56 @@
+import { ElementType, DetailedHTMLProps, HTMLAttributes, ComponentPropsWithRef } from 'react';
+import { ChildrenProps, StyleProps, ClickEvent } from './shared';
+
+type OmittedComponentStyleProps = 'style' | 'className';
+
+export type ModalComposedDialogElementType = 'article' | 'form';
+
+export type DialogElementProps = DetailedHTMLProps<HTMLAttributes<HTMLDialogElement>, HTMLDialogElement>;
+export type ModalComposedBodyElementProps = DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>;
+export type ModalComposedHeaderElementProps = DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>;
+export type ModalComposedFooterElementProps = DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>;
+
+export type ModalComposedDialogHandlingProps = {
+  isOpen: boolean;
+  onClose: (event: ClickEvent) => void;
+};
+
+export type ModalComposedDialogBaseProps<E extends ElementType = ModalComposedDialogElementType> = {
+  elementType?: E;
```

Comment:
done

Reply:
I see no change here.

Reply:
I had to change back, because the type is not properly right. I need to figure out where the binding to those selected element types is lost. Now if I set it as `elementType="div"` it should underline it. In addition, when I have `ElementTypeProps` there, it doesn't whisper to me that I can only use `article` or `form`. Instead, I only have a `div` (which is set as default in that type) or just a `JSX.constructor`.